src/App.tsx

```diff
@@ -9,34 +9,89 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+import { getPosts } from './utils/Post';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
```

Comment:
Why do you need it? Looks like it is redundant, you already have the whole selected user with ID in state.